walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,71 @@
 
 package com.google.sps;
 
+import static java.util.Collections.EMPTY_LIST;
+import static java.util.Collections.disjoint;
+import static java.util.Collections.sort;
+import static java.util.Arrays.asList;
+
+import java.util.Comparator;
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.List;
+
 
+// FindMeetingQuery finds open meeting timeslots throughout the day.
 public final class FindMeetingQuery {
+  /** 
+  *  @param events Set of events that attendees have, that need to be avoided.
+  *  @param request The duration of requested meeting and people attending. 
+  *  @return A list of open meeting timeslots.  
+  */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+      List<Event> eventsList = new ArrayList<>(events);
+      sort(eventsList, new Comparator<Event>() {
```

Comment:
I'm guessing your algorithm requires that you sort?  You might want a few sentences explaining what your algorithm is in your java doc.  (Most method-level java docs talk about what the method does, irrespective of implementation, but sometimes if the implementation is complicated enough, you might want a few more sentences.)

Reply:
I decided to add more java docs throughout the method.  I think will be easier to describe the algorithm when it is happening in the function.